src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -79,7 +73,7 @@
      * <p>This value is updated on each {@link net.dv8tion.jda.core.events.message.MessageReceivedEvent MessageReceivedEvent}
      * and <u><b>will be reset to {@code null} if the message associated with this ID gets deleted</b></u>
      *
-     * @throws java.lang.IllegalStateException
+     * @throws java.util.NoSuchElementException
```

Comment:
The documentation for `NoSuchElementException` suggests that it can be used outside of collections.

>Thrown by various accessor methods to indicate that the element being requested does not exist.

Reply:
Yes, but like I said, this isn't an element, it is an attribute.
Either way, the use of "Element" makes one think of an Iterator or Collection, which is why I believe IllegalStateException is better, especially since it quite literally is a state. A channel could have had a LastMessage a moment ago, but the state changed when the last message in the channel was deleted.